src/ucp/core/ucp_worker.c

```diff
@@ -1266,7 +1288,7 @@ void ucp_worker_iface_cleanup(ucp_worker_iface_t *wiface)
     ucp_worker_iface_disarm(wiface);
 
     if ((wiface->event_fd != -1) &&
-        (wiface->attr.cap.event_flags & UCT_IFACE_FLAG_EVENT_FD)) {
+        ucp_worker_iface_use_event_fd(wiface)) {
```

Comment:
maybe replace the check ucp_worker_iface_use_event_fd(wiface) by ucs_assert()?

Reply:
done